modules/magniteAnalyticsAdapter.js

```diff
@@ -633,6 +658,47 @@ const handleBidWon = args => {
   addEventToQueue({ bidsWon: [bidWon] }, bidWon.renderAuctionId, 'bidWon');
 }
 
+window.addEventListener('dmWebVitals', eventObj => {
+  let dmWebVitals = eventObj.detail;
+  dmWebVitals.accountId = accountId;
+  addEventToQueue({ dmWebVitals }, dmWebVitals.auctionId, 'webVital');
+});
+
+const countPendingEvents = (missedEvents, pendingEvents) => {
+  Object.keys(pendingEvents).forEach(event => {
+    if (Array.isArray(pendingEvents[event] && pendingEvents[event].length)) {
+      missedEvents[event] = pendingEvents[event].length;
+    } else if (event === 'dmWebVitals') {
+      missedEvents[event] = pendingEvents[event];
+    }
+  })
+}
+
+window.addEventListener('beforeunload', function () {
+  // Send auciton stuff
+  logInfo(`${MODULE_NAME}: Before Unload`);
+  let missedEvents = {};
+  Object.keys(cache.auctions).forEach(aid => {
+    if (!cache.auctions[aid].sent) {
+      missedEvents.auctions = missedEvents.auctions || 0;
+      missedEvents.auctions += 1;
+      if (cache.auctions[aid].pendingEvents && Object.keys(cache.auctions[aid].pendingEvents).length) {
+        countPendingEvents(missedEvents, cache.auctions[aid].pendingEvents);
+      }
+      logInfo(`${MODULE_NAME}: Auction ${aid} not sent!`);
+    }
+  });
+  countPendingEvents(missedEvents, cache.pendingEvents);
```

Comment:
So, in the case there are events that have not been sent yet, this function is executed twice? Why the second time? why not just executing it once? I am missing something about the logic here

Reply:
pendingEvents can either be at:

- Top level of cache: Meaning they are events which are not bound to an auction
   - `cache.pendingEvents`
- Ties to Auction Cache: Meaning they occured before the auction event fired, so they are together there
   - `cache.auctions[x].pendingEvents`

So we must check both of these places